### PR TITLE
config cache problems in problem api

### PR DESCRIPTION
**Pull request by reinsch82**
Initial version

some tests need fixing

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by reinsch82** [comment]
@bot-gradle test this

**Comment by reinsch82** [comment]
@bot-gradle test this

**Comment by reinsch82** [comment]
@bot-gradle test this

**Comment by reinsch82** [comment]
@bot-gradle test APT

File: platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/problems/ConfigurationCacheProblems.kt [hunk 1]
```diff
@@ -152,10 +161,54 @@ class ConfigurationCacheProblems(
...
 
+    private
+    fun InternalProblems.onProblem(problem: PropertyProblem, severity: ProblemSeverity) {
+        val message = problem.message.toString()
+        internalReporter.create {
+            id(
+                "configuration-cache-" + DeprecationMessageBuilder.createDefaultDeprecationId(message),
+                message,
+                GradleCoreProblemGroup.validation()
+            ) // TODO (reinhold) display name might be too elaborate
```
Comment (donat) [inline platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/problems/ConfigurationCacheProblems.kt:177]: Please double-check that. The id should be just a few words. We have the `contextualLabel` and the `details` fields to add longer descriptions.

File: testing/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/problems/KnownProblemIds.groovy [hunk 2]
```diff
@@ -118,6 +121,7 @@ class KnownProblemIds {
         'validation:type-validation:ignored-annotations-on-method' : 'Ignored annotations on method',
         'validation:type-validation:invalid-use-of-type-annotation' : 'Incorrect use of type annotation',
         'validation:type-validation:not-cacheable-without-reason' : 'Not cacheable without reason',
+        'validation:configuration-cache-cannot-serialize-object-of-type-org-gradle-api-defaulttask-a-subtype-of-org-gradle-api-task-as-these-are-not-supported-with-the-configuration-cache' : 'cannot serialize object of type \'org.gradle.api.DefaultTask\', a subtype of \'org.gradle.api.Task\', as these are not supported with the configuration cache.',
```
Comment (donat) [inline testing/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/problems/KnownProblemIds.groovy:124]: Although the id definitions look a bit suspicious, they might be correct as-is. These IDs are indeed context-insensitive and indeed uniquely define a particular configuration cache problem.

**Review by donat — CHANGES_REQUESTED** [review]
Looks good to me, only the TODOs in the PR should be addressed.

File: platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/problems/ConfigurationCacheProblems.kt [hunk 3]
```diff
@@ -152,10 +161,52 @@ class ConfigurationCacheProblems(
...
+            documentOfProblem(problem)
+            locationOfProblem(problem)
+            severity(severity.toProblemSeverity())
+        }.also { internalReporter.report(it) }
+    }
+
+    private
+    fun ProblemSpec.documentOfProblem(problem: PropertyProblem) {
+        problem.documentationSection?.let {
+            documentedAt(Documentation.userManual("configuration_cache", it.anchor).toString()) // TODO (reinhold) not sure?
```
Comment (donat) [inline platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/problems/ConfigurationCacheProblems.kt:188]: ❌ I assume it's a leftover TODO.

**Review by donat — APPROVED** [review]

**Pull request merged**